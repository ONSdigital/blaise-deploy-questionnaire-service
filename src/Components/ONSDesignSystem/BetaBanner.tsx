import React, {ReactElement} from "react";

function BetaBanner(): ReactElement {
    return (
        <>
            <div className="phase-banner">
                <div className="container ">
                    <div className="grid grid--flex grid--gutterless grid--vertical-center grid--no-wrap">
                        <div className="grid__col col-auto u-flex-no-grow">
                            <h3 className="phase-banner__badge">BETA</h3>
                        </div>
                        <div className="grid__col col-auto u-flex-shrink">
                            <p className="phase-banner__desc u-fs-s u-mb-no">This is a new service â€“ your <a
                                href="https://ons.service-now.com/">feedback</a> will help us improve it.</p>
                        </div>
                    </div>
                </div>
            </div>
        </>
    );
}

export default BetaBanner;
