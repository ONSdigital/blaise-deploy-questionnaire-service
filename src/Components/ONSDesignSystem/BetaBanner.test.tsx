import React from "react";
import Enzyme, {shallow, render} from "enzyme";

import Adapter from "enzyme-adapter-react-16";
import BetaBanner from "./BetaBanner";

describe("ONS In Dev Banner Test", () => {
    Enzyme.configure({ adapter: new Adapter() });

    it("matches Snapshot", () => {
        expect(render(<BetaBanner/>)).toMatchSnapshot();
    });

    it("should render correctly", () => expect(shallow(<BetaBanner/>).exists()).toEqual(true));
});