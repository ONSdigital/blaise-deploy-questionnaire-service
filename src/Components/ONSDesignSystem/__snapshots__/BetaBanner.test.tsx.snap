// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ONS In Dev Banner Test matches Snapshot 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "phase-banner",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "container ",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "grid grid--flex grid--gutterless grid--vertical-center grid--no-wrap",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "class": "grid__col col-auto u-flex-no-grow",
                },
                "children": Array [
                  Object {
                    "attribs": Object {
                      "class": "phase-banner__badge",
                    },
                    "children": Array [
                      Object {
                        "data": "BETA",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                    ],
                    "name": "h3",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                    },
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": Object {
                  "attribs": Object {
                    "class": "grid__col col-auto u-flex-shrink",
                  },
                  "children": Array [
                    Object {
                      "attribs": Object {
                        "class": "phase-banner__desc u-fs-s u-mb-no",
                      },
                      "children": Array [
                        Object {
                          "data": "This is a new service – your ",
                          "next": Object {
                            "attribs": Object {
                              "href": "https://ons.service-now.com/",
                            },
                            "children": Array [
                              Object {
                                "data": "feedback",
                                "next": null,
                                "parent": [Circular],
                                "prev": null,
                                "type": "text",
                              },
                            ],
                            "name": "a",
                            "namespace": "http://www.w3.org/1999/xhtml",
                            "next": Object {
                              "data": " will help us improve it.",
                              "next": null,
                              "parent": [Circular],
                              "prev": [Circular],
                              "type": "text",
                            },
                            "parent": [Circular],
                            "prev": [Circular],
                            "type": "tag",
                            "x-attribsNamespace": Object {
                              "href": undefined,
                            },
                            "x-attribsPrefix": Object {
                              "href": undefined,
                            },
                          },
                          "parent": [Circular],
                          "prev": null,
                          "type": "text",
                        },
                        Object {
                          "attribs": Object {
                            "href": "https://ons.service-now.com/",
                          },
                          "children": Array [
                            Object {
                              "data": "feedback",
                              "next": null,
                              "parent": [Circular],
                              "prev": null,
                              "type": "text",
                            },
                          ],
                          "name": "a",
                          "namespace": "http://www.w3.org/1999/xhtml",
                          "next": Object {
                            "data": " will help us improve it.",
                            "next": null,
                            "parent": [Circular],
                            "prev": [Circular],
                            "type": "text",
                          },
                          "parent": [Circular],
                          "prev": Object {
                            "data": "This is a new service – your ",
                            "next": [Circular],
                            "parent": [Circular],
                            "prev": null,
                            "type": "text",
                          },
                          "type": "tag",
                          "x-attribsNamespace": Object {
                            "href": undefined,
                          },
                          "x-attribsPrefix": Object {
                            "href": undefined,
                          },
                        },
                        Object {
                          "data": " will help us improve it.",
                          "next": null,
                          "parent": [Circular],
                          "prev": Object {
                            "attribs": Object {
                              "href": "https://ons.service-now.com/",
                            },
                            "children": Array [
                              Object {
                                "data": "feedback",
                                "next": null,
                                "parent": [Circular],
                                "prev": null,
                                "type": "text",
                              },
                            ],
                            "name": "a",
                            "namespace": "http://www.w3.org/1999/xhtml",
                            "next": [Circular],
                            "parent": [Circular],
                            "prev": Object {
                              "data": "This is a new service – your ",
                              "next": [Circular],
                              "parent": [Circular],
                              "prev": null,
                              "type": "text",
                            },
                            "type": "tag",
                            "x-attribsNamespace": Object {
                              "href": undefined,
                            },
                            "x-attribsPrefix": Object {
                              "href": undefined,
                            },
                          },
                          "type": "text",
                        },
                      ],
                      "name": "p",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "class": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "class": undefined,
                      },
                    },
                  ],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              Object {
                "attribs": Object {
                  "class": "grid__col col-auto u-flex-shrink",
                },
                "children": Array [
                  Object {
                    "attribs": Object {
                      "class": "phase-banner__desc u-fs-s u-mb-no",
                    },
                    "children": Array [
                      Object {
                        "data": "This is a new service – your ",
                        "next": Object {
                          "attribs": Object {
                            "href": "https://ons.service-now.com/",
                          },
                          "children": Array [
                            Object {
                              "data": "feedback",
                              "next": null,
                              "parent": [Circular],
                              "prev": null,
                              "type": "text",
                            },
                          ],
                          "name": "a",
                          "namespace": "http://www.w3.org/1999/xhtml",
                          "next": Object {
                            "data": " will help us improve it.",
                            "next": null,
                            "parent": [Circular],
                            "prev": [Circular],
                            "type": "text",
                          },
                          "parent": [Circular],
                          "prev": [Circular],
                          "type": "tag",
                          "x-attribsNamespace": Object {
                            "href": undefined,
                          },
                          "x-attribsPrefix": Object {
                            "href": undefined,
                          },
                        },
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                      Object {
                        "attribs": Object {
                          "href": "https://ons.service-now.com/",
                        },
                        "children": Array [
                          Object {
                            "data": "feedback",
                            "next": null,
                            "parent": [Circular],
                            "prev": null,
                            "type": "text",
                          },
                        ],
                        "name": "a",
                        "namespace": "http://www.w3.org/1999/xhtml",
                        "next": Object {
                          "data": " will help us improve it.",
                          "next": null,
                          "parent": [Circular],
                          "prev": [Circular],
                          "type": "text",
                        },
                        "parent": [Circular],
                        "prev": Object {
                          "data": "This is a new service – your ",
                          "next": [Circular],
                          "parent": [Circular],
                          "prev": null,
                          "type": "text",
                        },
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "href": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "href": undefined,
                        },
                      },
                      Object {
                        "data": " will help us improve it.",
                        "next": null,
                        "parent": [Circular],
                        "prev": Object {
                          "attribs": Object {
                            "href": "https://ons.service-now.com/",
                          },
                          "children": Array [
                            Object {
                              "data": "feedback",
                              "next": null,
                              "parent": [Circular],
                              "prev": null,
                              "type": "text",
                            },
                          ],
                          "name": "a",
                          "namespace": "http://www.w3.org/1999/xhtml",
                          "next": [Circular],
                          "parent": [Circular],
                          "prev": Object {
                            "data": "This is a new service – your ",
                            "next": [Circular],
                            "parent": [Circular],
                            "prev": null,
                            "type": "text",
                          },
                          "type": "tag",
                          "x-attribsNamespace": Object {
                            "href": undefined,
                          },
                          "x-attribsPrefix": Object {
                            "href": undefined,
                          },
                        },
                        "type": "text",
                      },
                    ],
                    "name": "p",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                    },
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": Object {
                  "attribs": Object {
                    "class": "grid__col col-auto u-flex-no-grow",
                  },
                  "children": Array [
                    Object {
                      "attribs": Object {
                        "class": "phase-banner__badge",
                      },
                      "children": Array [
                        Object {
                          "data": "BETA",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "text",
                        },
                      ],
                      "name": "h3",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "class": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "class": undefined,
                      },
                    },
                  ],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
